### ISSUE TITLE
ENLACE never leaves the front's default: the connection event is emitted before the window listens

### ISSUE THREAD

**Issue by Foxandxss** [Foxandxss/modx-analyzer#18]
Found with the real MODX8 in front of it, during #13's keyboard session.

## What is on screen

The dev readout says, for a whole 12-minute session:

```
ENLACE  DESCONECTADO · — · ÚLTIMO SONDEO 0.0 s
```

and the connection dot beside `MODX-1` in the header is dark, all session.

Meanwhile the same screen shows `RELECTURA 383 DE 384 · 1,76 s`, an ancla beating at 1 Hz, the anillo ancho at 7,9 Hz and a full diagram of `FM CS80 Brass`. The keyboard is answering 383 addresses. The app says the port is gone.

## Why it is not a watchdog misfire

`DESCONECTADO` with `loss` drawn as a **dash** cannot come from the native side. Every `Link::Disconnected` carries a reason:

- `connection.rs:73` — `Link::Disconnected(loss) => { view.loss = Some(loss.as_str()); }`
- `keyboard.rs:117`, `keyboard.rs:229` — both pass `Loss::Enumeration`

`port: "disconnected"` **with** `loss: null` is exactly one thing: the front's untouched constructor default at `ui/src/app/backend/tauri-backend-gateway.ts:120`. So the webview has never received a single `modx://connection`.

## The cause

`Keyboard::connect` emits `Link::Connected` from the setup thread (`keyboard.rs:96`), very likely before the window has run a line of JavaScript. `LinkWatch` then only re-emits **on a change** (`link.rs:88`), and nothing changes, so the one missed event is missed for the rest of the session.

This is the same trap the gateway already guards for two other events, in the same constructor:

```ts
// The volcado is taken once, on a thread, and is very likely finished before
// this window has run a line of JavaScript. The event would then have nobody
// listening, so the state is asked for as well as listened to.
void invoke<DumpView | null>('last_dump')...
void invoke<GeneratorView>('generator_state')...
```

There is no `connection_state` command to do the same for the link.

## Suggested fix

A `connection_state` command returning the merged `ConnectionView` from `Connection`'s mutex, invoked alongside `last_dump` and `generator_state`. `Connection` already holds the state and already has the lock; only the getter and the command are missing.

## Why it blocks work

- **#15 cannot be verified at all.** Its whole acceptance criterion is pulling the USB cable and recording what `ENLACE` and the card said. The line has been wrong since launch, so nothing read off it is a result.
- The `DIAGRAMA CONGELADO` card correctly does **not** draw, because it hangs off the *reason* and not the state (a deliberate decision in `unhappy-cards.ts`). So the bug is invisible on the main screen and only the dev line and the dark dot give it away.

## Not affected

#13's criteria. The ancla, the relectura, the invalidation and the header states all work off other events and were verified with this bug present.

**Comment by Foxandxss** [comment]
The 60 fps capture from #13's session adds one symptom: **`ÚLTIMO SONDEO` goes negative.**

`ÚLTIMO SONDEO -0.1 s` appears repeatedly across the run (t = 11.0 s and t = 11.7 s among others), alternating with `0.0` and `0.1`.

```
ENLACE  DESCONECTADO · — · ÚLTIMO SONDEO -0.1 s
```

It comes from `dev-readout.ts:163`:

```ts
const since = at === null ? DEAD_MARK : `${((this.clock.now() - at) / 1000).toFixed(1)} s`;
```

`lastPollAt` returns a `readAt` aligned from the wire's *ages* onto this side's clock, and that alignment can land slightly ahead of `clock.now()`. Clamping at zero would hide it; the real point is that the whole line is being drawn for a port that is open and answering, which is this issue.

Worth fixing together, since `ÚLTIMO SONDEO N s` is also the card's own sentence in #15 and a negative age on a card is worse than on a dev line.

**Comment by Foxandxss** [comment]
Built and committed in `62aaff3`. **Leaving this open**: it needs one glance at the running app with the keyboard plugged in, and that has not happened.

## Done

- **`connection_state`**, in `src-tauri/src/connection.rs`, returning the merged `ConnectionView` from the mutex `Connection` already holds. `Connection::view()` is the getter, and `is_disconnected` now goes through it too rather than taking the lock a second way.
- **The front asks as well as listens**, in the gateway constructor — the same guard `last_dump` and `generator_state` already had, for the third of the three. It applies the answer **only if the event has not already arrived**, so a slow reply cannot overwrite a newer state:

  ```ts
  let heardConnection = false;
  this.listen<ConnectionView>('modx://connection', (view) => {
    heardConnection = true;
    this.connection.set(view);
  });
  void invoke<ConnectionView>('connection_state').then((state) => {
    if (!heardConnection) {
      this.connection.set(state);
    }
  });
  ```

- **`ÚLTIMO SONDEO` can no longer go negative.** The clamp moves into `pollAgeSeconds` in `provenance/last-poll.ts` and both printers use it. The card was already clamping and the dev readout was not — one clamp in two places is the kind of thing that drifts, and it had already drifted.
- **`rereadLine`'s comment now says what was measured** rather than what the design expected: 1,7 s and 383 of 384, playing and idle.

159 tests in the front (2 new, on the clamp), 123 in `modx-midi`, 17 in the app crate, 48 in `modx-dsp`, 15 in `modx-audio`; lint, test and build green.

## Not done — why this stays open

- [ ] **`ENLACE` says `MODX-1 · ABIERTO` with the keyboard plugged in, and the dot beside `MODX-1` is lit.** Nobody has launched the app since the change.

`tauri-backend-gateway.ts` is the only file in the front that touches `@tauri-apps/api` and it has no tests. Building a mock harness for five lines of guard would be worse than the guard, so this one is verified by launching the app and not by a test — which is the same way `HardwarePort::open` is verified, and worth being explicit about rather than implying the green suite covers it.

## Worth knowing

The fix removes the reason `ENLACE` was wrong at rest. It does **not** touch what the line says once the watchdog has looked, so #15's cable-pull measures exactly what it was always meant to measure — it just now starts from a line that is true.

**Comment by Foxandxss** [comment]
Verified on the running app with the MODX8 plugged in:

```
ENLACE   MODX-1 · ABIERTO
```

The connection dot beside `MODX-1` is lit, and no `DESCONECTADO` anywhere. Fixed in `62aaff3`.

**#15 is unblocked** — the cable-pull now starts from a line that is true at rest, so what it measures is the watchdog and not the startup race.
